doc/design/katib_on_sqlflow.md

```diff
@@ -1,70 +1,91 @@
+## Katib
 
-## Hyperparameter Optimization in Katib
+[Katib](https://github.com/kubeflow/katib) is a Kubernetes Native System for Hyperparameter Tuning and Neural Architecture Search.  The inspiration of Katib comes from Google Vizier and supports multiple machine learning frameworks, for example, TensorFlow, Apache MXNet, PyTorch, and XGBoost.  We compared Katib with some other auto hyperparameter tunning systems, and we prefer its Kubernetes-native architecture.
 
-[Katib](https://github.com/kubeflow/katib) is a Kubernetes Native System for Hyperparameter Tuning and Neural Architecture Search. The system is inspired by Google Vizier and supports multiple ML/DL frameworks (e.g. TensorFlow, Apache MXNet, PyTorch and XGBoost).
+However, Katib, or hyperparameter tuning in the academic literature, is not sufficient for our use case.
 
-A training job in Katib is called an experiment. Users need to specify or choose: 
-- parameters need to be optimized in the training. 
-- the value range for each parameter.
-- the algorithm (e.g., `random`) used to choose candidate value for training paramters.
-- objective (e.g., accuracy) which measures the model with different parameters' value.
+## The Paradox
 
-When an experiment is done, users can check objectives of each set of training parameters' value via web UI. Then users can decide the best value for those training parameters.
+To define a training job, a.k.a., *experiment*, in Katib, users need to specify the search range of each hyperparameter.
 
-## Couler and SQLFlow
+Ironically, it is an extra burden for the users to specify the above information, as our goal is to release users from specifying hyperparameters.
 
-SQLFlow does not create training jobs on Katib directly. Instead, SQLFlow creates Katib steps in a workflow via couler, and the workflow will be run by Argo on Kubernetes. More details can be found in [couler and SQLFlow](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/design/couler_sqlflow.md). 
+## Untangle the Paradox
 
-## SQLFlow Syntax
+### Boosting Tree Models
 
-In order to use Katib, programmers need to specify it is a Katib job by `TRAIN katib.{tf, pytorch, xgboost}.{...}` in SQL query. Here we use a simple example to explain the syntax of creating Katib job for XGBoost training in SQLFlow.  
+For boosting tree models, especially models with XGBoost, there is a small group of effective hyperparameter, and we can empirically determine their ranges.  We noticed that the following two are the most important.
 
-``` sql
-SELECT * FROM train_table
-TO TRAIN a_data_scientist/xgboost:v0.5/katib.xgboost.gbtree
+- `max_depth` in the range [2,10], and
+- `num_round` in the range [50, 100].
+
+With the introduction of auto hyperparameter tunning, we hope that users don't need to specify the `num_round` and `max_depth` values in the following SQL statement.
+
+```sql
+SELECT * FROM a_dataset_table
+TO TRAIN a_data_scientist/xgboost_models:v0.5/a_gbtree_model
```

Comment:
I realize there are potential problems to allow programmers to use their own image in katib tuning job.
1. The collector in Katib requires the job container's output to follow a certain format, like `accuracy = 0.97`. And users need to tell Katib to collect data for `accuracy` in katib job yaml.
2. Katib does not obtain the tuned model. When one pod finishes tuning, it needs to write the tuned model to a store, e.g., odps, then the following pred can make use of it.

I think it requires too much on programmers in their own image. Maybe sqlflow invokes submitters to train model when programmers specify the image; otherwise, sqlflow uses katib.